PC/launcher2.c

```diff
@@ -826,6 +841,72 @@ searchPath(SearchInfo *search, const wchar_t *shebang, int shebangLength)
         return RC_BAD_VIRTUAL_PATH;
     }
 
+    // Make sure we didn't find a reparse point that will open the Microsoft Store
+    // If we did, pretend there was no shebang and let normal handling take over
+    WIN32_FIND_DATAW findData;
+    HANDLE hFind = FindFirstFileW(buffer, &findData);
+    if (!hFind) {
+        if (GetLastError() == ERROR_FILE_NOT_FOUND) {
+            debug(L"# %s on disappeared\n", buffer);
+            // The file found on PATH disappeared. Alias probably disabled by user while trying to run, let normal handling take over
+            RC_NO_SHEBANG;
+        }
+
+        // Other errors should cause us to break
+        winerror(0, L"Failed to find %s on PATH\n", filename);
+        return RC_BAD_VIRTUAL_PATH;
+    }
+
+    if (findData.dwFileAttributes & FILE_ATTRIBUTE_REPARSE_POINT &&
+        findData.dwReserved0 & IO_REPARSE_TAG_APPEXECLINK) {
+        const size_t bufSize = sizeof(AppExecLinkFile);
+        wchar_t appExecLinkBuf[sizeof(AppExecLinkFile)];
+
+        HANDLE hReparsePoint = CreateFileW(buffer, 0, FILE_SHARE_READ, NULL, OPEN_EXISTING, FILE_FLAG_OPEN_REPARSE_POINT, NULL);
+        if (!hReparsePoint) {
+            if (GetLastError() == ERROR_FILE_NOT_FOUND) {
+                debug(L"# %s on disappeared\n", buffer);
+                // The file found on PATH disappeared. Alias probably disabled by user while trying to run, let normal handling take over
+                RC_NO_SHEBANG;
+            }
+
+            // Other errors should cause us to break
+            winerror(0, L"Failed to find %s on PATH\n", filename);
+            return RC_BAD_VIRTUAL_PATH;
+        }
+
+        if (!DeviceIoControl(hReparsePoint, FSCTL_GET_REPARSE_POINT, NULL, 0, appExecLinkBuf, bufSize, NULL, NULL)) {
+            if (GetLastError() == ERROR_FILE_NOT_FOUND) {
+                debug(L"# %s on disappeared\n", buffer);
+                // The file found on PATH disappeared. Alias probably disabled by user while trying to run, let normal handling take over
+                RC_NO_SHEBANG;
+            }
```

Comment:
Ditto - no benefit to calling this one out separately. Normal handling for all errors should be okay